### PR TITLE
Fix issue where out of sync packs caused a crash

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27997

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 1]
```diff
@@ -103,13 +104,27 @@ def _does_pack_metadata_exist(self):
...
+                return True
+            elif LooseVersion(master_current_version) > LooseVersion(new_version):
+                print_error("The master branch is currently ahead of your pack's version. "
+                            "Please pull from master and re-run the command.")
+                sys.exit(0)
+            elif LooseVersion(master_current_version) < LooseVersion(new_version):
+                return False
         except RuntimeError:
-            print_warning(f"Unable to locate a pack with the name {self.pack} in the git diff. "
+            print_warning(f"Unable to locate a pack with the name {self.pack} in the git diff.\n"
```
Comment (bakatzir) [inline demisto_sdk/commands/update_release_notes/update_rn.py:127]: Don't we want to sys.exit(0) in that case?

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 2]
```diff
@@ -103,13 +104,27 @@ def _does_pack_metadata_exist(self):
 
     def is_bump_required(self):
```
Comment (bakatzir) [inline demisto_sdk/commands/update_release_notes/update_rn.py:105]: can you add a docstring?
It is hard to follow the logic(at least for me). e.g: what this phrase means `if len(master_metadata) > 0:`

File: demisto_sdk/commands/update_release_notes/tests/update_rn_test.py [hunk 3]
```diff
@@ -556,3 +556,40 @@ def test_execute_update_invalid(self):
         update_rn = UpdateRN(pack_path="Packs/Legacy", update_type='minor', pack_files=set(),
                              added_files=set())
         update_rn.execute_update()
+
+    diff_package = [
+        ("1.0.1", "1.0.2", True),
+        ("1.0.5", "1.0.4", False),
+        ("1.0.5", "1.0.5", True)
+    ]
```
Comment (bakatzir) [inline demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:564]: nice UT.
can you also cover the branch of of: `len(master_metadata) > 0:`?

Comment (amshamah419) [reply demisto_sdk/commands/update_release_notes/tests/update_rn_test.py:564]: In this case the `master_current_version` gets set to '99.99.99'. The UT is difficult since the mock patch function isn't allowing the variable to be redeclared within the unit test.

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 2 above)
Comment (amshamah419) [reply demisto_sdk/commands/update_release_notes/update_rn.py:105]: That's just a check that the metadata exists. If the length is longer than 0, we know it can be parsed.

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -102,15 +103,35 @@ def _does_pack_metadata_exist(self):
...
+                return True
+            elif LooseVersion(master_current_version) > LooseVersion(new_version):
+                print_error("The master branch is currently ahead of your pack's version. "
+                            "Please pull from master and re-run the command.")
+                sys.exit(0)
+            elif LooseVersion(master_current_version) < LooseVersion(new_version):
+                return False
         except RuntimeError:
-            print_warning(f"Unable to locate a pack with the name {self.pack} in the git diff. "
+            print_warning(f"Unable to locate a pack with the name {self.pack} in the git diff.\n"
```
Comment (bakatzir) [inline demisto_sdk/commands/update_release_notes/update_rn.py:132]: I guess a print_error than

**Review by bakatzir — APPROVED** [review]
